modules/javafx.graphics/src/main/java/javafx/scene/Node.java

```diff
@@ -1364,7 +1364,30 @@ public final boolean isVisible() {
     public final BooleanProperty visibleProperty() {
         if (visible == null) {
             visible = new StyleableBooleanProperty(true) {
+                static final CssMetaData<Node, Boolean> METADATA = new CssMetaData<>(
```

Comment:
One problem: this idea requires creation of an instance, bc METADATA static objects get created when an instance is created.
I'd rather avoid this inversion...

Reply:
We're fully inflating a single instance of each particular class once to gather all metadata. After that, all further instances don't need to be fully inflated.

Reply:
what if this is an abstract custom Node/Control?

Reply:
We would never expect to encounter an instance of an abstract class. Generally, metadata reflection is done the first time we call `getCssMetaData` for any particular class, which we can only do if we have an instance of that class.

The only exceptions to this are the preexisting static `getClassCssMetaData` methods, which are initialized once with a throwaway instance of the class they are declared in, for example in the case of the `Node` class:
```java
private static final List<CssMetaData<? extends Styleable, ?>> classCssMetaData =
    CssMetaData.of(new Node() {});
```

New classes should not copy this pattern of having a static `getClassCssMetaData` method, so the problem shouldn't exist for these classes.